src/main/java/com/ericsson/esj/JenkinsManager.java

```diff
@@ -0,0 +1,259 @@
+     * script added.
+     *
+     * @param job_token
+     * @param script
+     * @return
+     * @throws IOException
+     */
+    public String getXmlJobData(String job_token, String script) throws IOException {
+        String jobData = getStringFromFile(JENKINS_JOB_TEMPLATE_FILE_PATH);
+
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+        if (script == null) {
+            script = DEFAULT_SCRIPT;
+        }
+
+        jobData = injectScriptAndTokenIntoXmlString(jobData, job_token, script);
+        return jobData;
+    }
+
+    /**
+     * Creates a jenkins job with a given name using the XML data as input for
+     * job configuration
+     *
+     * @param jobName
+     * @param jobXmlData
+     * @return
+     * @throws URISyntaxException
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public boolean createJob(String jobName, String jobXmlData) throws URISyntaxException {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        boolean success = false;
+
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_XML).addHeader("Jenkins-Crumb", crumb)
+                .addParam("name", jobName).setBody(jobXmlData).setEndpoint("/createItem");
+
+        ResponseEntity response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.SC_OK;
+
+        if(!success) {
+            LOGGER.error("Failed to create a jenkins job. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
+        }
+
+        return success;
+    }
+
+    /**
+     * This function recieves a jenkins job name and are used to trigger the job
+     *
+     * @throws URISyntaxException
+     */
+    public boolean triggerJob(String jobName, String job_token) throws URISyntaxException {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+        boolean success = false;
+
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+
+        String endpoint = "/job/" + jobName + "/build";
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON).addParam("token", job_token)
+                .setEndpoint(endpoint);
+
+        ResponseEntity response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.SC_CREATED;
+
+        if(!success) {
+            LOGGER.error("Failed to trigger a jenkins job. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
```

Comment:
Maybe this should also throw an error if it did not succeed? Then the user may or may not handle it depending on their implementation